src/lib.rs

```diff
@@ -84,30 +72,539 @@ fn serialize_from_raw_pointer<T: ToBytes>(
         .expect(format!("unable to write {} to buffer", type_name::<T>()).as_str())
 }
 
-fn deserialize_from_file<T: FromBytes>(
-    file_path: &Path,
-) -> Option<T> {
-    match read_from_file(file_path) {
-        Ok(t) => Some(t),
+pub fn free_pointer<T> (ptr: *mut T) {
+    if ptr.is_null() { return };
+
+    unsafe { drop( Box::from_raw(ptr)) }
+}
+
+//*********** Commitment Tree functions ****************
+use cctp_primitives::commitment_tree::CommitmentTree;
+use cctp_primitives::commitment_tree::hashers::hash_bytes;
+
+#[no_mangle]
+pub extern "C" fn zendoo_commitment_tree_create() -> *mut CommitmentTree {
+
+    let cmt = CommitmentTree::create();
+    let heap_obj = Box::new(cmt);
+    println!("Ctor: heap_obj {:p}", heap_obj);
+    let raw_obj = Box::into_raw(heap_obj);
+    println!("      raw_obj  {:p}", raw_obj);
+    raw_obj
+}
+
+#[no_mangle]
+pub extern "C" fn zendoo_commitment_tree_delete(ptr : *mut CommitmentTree) {
+    if ptr.is_null() {
+        println!("delete: nullptr!");
+        return;
+    }
+    unsafe {
+        println!("Dtor: raw_obj  {:p}", ptr);
+        let heap_obj = Box::from_raw(ptr);
+        println!("      heap_obj {:p}", heap_obj);
+    }
+}
+
+
+#[no_mangle]
+pub extern "C" fn zendoo_commitment_tree_add_scc(ptr : *mut CommitmentTree,
+    sc_id: *mut BufferWithSize,
```

Comment:
Here as well as in further occurrences where `*mut BufferWithSize` parameter is used for CommitmentTree-related wrappers, `mut` is redundant, due to the underlying methods of CommitmentTree take such byte arrays as immutable parameters

Reply:
ok, done